src/core/qgsofflineediting.cpp

```diff
@@ -164,157 +165,189 @@ void QgsOfflineEditing::synchronize()
   const QgsSnappingConfig snappingConfig = QgsProject::instance()->snappingConfig();
 
   // restore and sync remote layers
-  QList<QgsMapLayer *> offlineLayers;
   QMap<QString, QgsMapLayer *> mapLayers = QgsProject::instance()->mapLayers();
+  QMap<int, std::shared_ptr<QgsVectorLayer>> remoteLayersByOfflineId;
```

Comment:
Are these layers used after this method exits or only in here?
I think they could safely be created as `std::unique_ptr<>` if only used in here.

Reply:
I am afraid it is better we keep it as `std::shared_ptr<>` https://stackoverflow.com/a/60936263/2319028

Reply:
Makes sense, thanks!